File: DSA/Rakshit-Singh/Day 5/Day_5_Question_1.java
class Day_5_Question_1
{
    public static String reverse(Node head){
        String s="";
        while(head!=null){
            s=head.data + " "+s;
            head=head.next;
        }
        return s;
    }
    public static void main(String args[]){
        Node head = new Node(1);
        head.next=new Node(2);
        head.next.next= new Node(3);
        System.out.println(reverse(head));
    }
}

File: DSA/Vikash_kumar/Day_5_Question_1.py
#Day5-Question1-
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None
    
    def append(self, data):
        new_node = Node(data)
        if not self.head:
            self.head = new_node
            return
        last = self.head
        while last.next:
            last = last.next
        last.next = new_node

    def print_reverse(self):
        stack = []
        current = self.head
        while current:
            stack.append(current.data)
            current = current.next
        while stack:
            print(stack.pop(), end=" ")

def create_linked_list():
    linked_list = LinkedList()
    elements = input("Enter the elements of the linked list separated by space: ").split()
    for element in elements:
        linked_list.append(int(element))
    return linked_list

linked_list = create_linked_list()

print("The elements of the linked list in reverse order are:")
linked_list.print_reverse()
